apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/EnumMap.cs

```diff
@@ -30,14 +30,26 @@ public ApiValueAttribute(string value)
             Value = value;
         }
 
-        internal static void ForEachStaticField<T>(Action<T, string> execute)
+        public static IEnumerable<ApiValueName<T>> GetApiValueNamesIn<T>() where T : struct => 
+            from field in typeof(T).GetTypeInfo().DeclaredFields
+            where field.IsStatic
+            select ApiValueName<T>.Create(field);
+
+        public struct ApiValueName<T> where T : struct
```

Comment:
I'm not sure this adds much, to be honest - the code you had before looked pretty simple, and we can just use `KeyValuePair` for name and value. `IntValue` is only used in one place, so that cast can be there.

Reply:
Rationale behind the struct:
I didn't like a lot the idea of using KeyValuePair given that
a) there's no specific ToDictionary() method for IEnumerable<KeyValuePair> that doesn't require the use of selectors (unless I'm missing an extension import)
b) our three dictionariaes have different key value pairs so the returned IEnumerable<KeyValuePair> can only match one of the dictionaries, which is confusing, i.e. receiving one KeyValuePair to turn it into another. (Unless we passed a function to build the KeyValuePair but then, because of a) we have to use selectors anyway, which seems redundant).
So I though of using a Tuple instead of a KeyValuePair. That sort of lead to the struct so as to encapsulate Creation and IntValue.
But, my heart is not set on the Tuple vs. struct question, I see that this is simple enough and just locally used so as to not require a specific struct. But what I really don't like is using KeyValuePair, I think is confusing and/or redundant (again unless I'm missing a ToDictionary() extension method that takes no selectors).
Happy to chat later.